portal/src/components/Materials/Materials.component.js

```diff
@@ -54,20 +51,46 @@ export default {
       }
     },
     /**
-     * Select material
+     * Select material from collection
      * @param material - {Object}
+     * @param selectFor - string
      */
-    selectMaterial(material) {
-      let selected_materials = this.value.slice(0)
-
-      if (selected_materials.indexOf(material.external_id) === -1) {
-        selected_materials.push(material.external_id)
+    selectMaterial(material, selectFor) {
+      if (selectFor === 'delete') {
+        this.deleteMaterial(material)
       } else {
-        selected_materials = selected_materials.filter(
-          item => item !== material.external_id
-        )
+        let selected_materials = this.value.slice(0)
+
+        if (selected_materials.indexOf(material.external_id) === -1) {
+          selected_materials.push(material.external_id)
```

Comment:
Misschien een method `toggleMaterial` maken, die het materiaal toevoegt of verwijderd. 

```
if (this.value.indexOf(material.external_id) === -1) {
    return [...this.value, material.external_id]
else {
    return this.value.filter(
        item => item !== material.external_id
    )
}
```

Dan kan hier `this.$emit('input', this.toggleMaterial(material))`

Dat zou het voor mij wat sneller leesbaarder maken wat er gebeurt

Reply:
ja het is niet zo duidelijk inderdaad